semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/DenotationOps.scala

```diff
@@ -122,30 +122,36 @@ trait DenotationOps { self: DatabaseOps =>
         if (saveOverrides && config.denotations.saveReferences) gsym.overrides
         else Nil
       }
-      config.signatures match {
-        case SignatureMode.None =>
-          val denot = m.Denotation(flags, name, "", Nil, Nil, over, None)
-          DenotationResult(denot, todoOverrides, Nil)
-        case SignatureMode.Old =>
-          val (signature, names) = oldInfo
-          val denot = m.Denotation(flags, name, signature, names, Nil, over, None)
-          DenotationResult(denot, todoOverrides, Nil)
-        case SignatureMode.New =>
-          val (tpe, todoTpe) = newInfo
-          val denot = m.Denotation(flags, name, "", Nil, Nil, over, tpe)
-          DenotationResult(denot, todoOverrides, todoTpe)
-        case SignatureMode.All =>
-          val (signature, names) = oldInfo
-          val (tpe, todoTpe) = newInfo
-          val denot = m.Denotation(flags, name, signature, names, Nil, over, tpe)
-          DenotationResult(denot, todoOverrides, todoTpe)
-      }
+      val todoAnnotations = gsym.annotations.map(_.symbol)
+      val annot = gsym.annotations.map(_.symbol.toSemantic)
+      val (denot, todoTpe) =
+        config.signatures match {
```

Comment:
Please try to minimize the diffs by moving `config.signatures` right after `=`. Scalafmt preserves the line break if you insert it